core/src/main/java/hudson/model/Queue.java

```diff
@@ -2337,6 +2337,33 @@ boolean leave(Queue q) {
         }
     }
 
+    /**
+     * Mock queue Item used for unit testing.
+     * <p/>
+     * No other clean way to effectively mock Item.
+     */
+    @SuppressWarnings("unchecked")
+    public static final class MockItem extends Item {
```

Comment:
It makes sense to move this code to test classes. No need to keep it in core/src/main IMO